next-desolates-map/components/SettingsPage/SettingsPage.tsx

```diff
@@ -0,0 +1,466 @@
+import { useContext, useState } from "react";
+import CameraMovement from "../../core/modules/CameraMovement";
+import { Clusters, SpaceRendererContext, Stages } from "../../pages/_app";
+import { searchForPlanetAndChangeStage } from "../RenderController/RenderController";
+import Image from "next/image"
+import pic from "../../public/profile.png"
+
+
+export default function SettingsPage() {
+    const {
+        changeStage,
+        p5,
+        cam,
+        setOngoingCamMov,
+        celestialObjects,
+        setCluster,
+        setSelectedPlanet,
+    } = useContext(SpaceRendererContext);
+
+    const [planetSearchInput, setPlanetSearchInput] = useState("");
+
+    function handleInputChange(e: any) {
+        setPlanetSearchInput(e.target.value);
+    }
+
+    function handleKeyDown(e: any) {
+        if (e.key === "Enter") {
+            console.log("ENTER PRESSED");
+            searchForPlanet();
+        }
+    }
+
+    function searchForPlanet() {
+        searchForPlanetAndChangeStage(
+            planetSearchInput,
+            p5!,
+            cam!,
+            setOngoingCamMov,
+            celestialObjects,
+            setCluster,
+            changeStage,
+            setSelectedPlanet
+        );
+    }
+
+    return (
+        // Global window
+     <div className="fixed top-7 right-0 left-0 flex mt-10 lg:absolute lg:flex lg:justify-center">
```

Comment:
I think prettify may not be working, mind checking? It seems like the indentation is a bit wrong in multiple places

Reply:
Done. Prettify wasn't working for this file particularly. Now should work. Thank you again for checking everything!